lib/llm/providers/gemini/error_handler.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class LLM::Gemini
+  class ErrorHandler
+    ##
+    # @return [Net::HTTPResponse]
+    #  Non-2XX response from the server
+    attr_reader :res
+
+    ##
+    # @param [Net::HTTPResponse] res
+    #  The response from the server
+    # @return [LLM::Gemini::ErrorHandler]
+    def initialize(res)
+      @res = res
+    end
+
+    ##
+    # @raise [LLM::Error]
+    #  Raises a subclass of {LLM::Error LLM::Error}
+    def raise_error!
+      case res
+      when Net::HTTPBadRequest
+        # FIXME:
+        # Parse the response body for more information
+        # Based on the request body, raise a more specific error
```

Comment:
We should just parse the response for the relevant error, now that we have per-provider error handling :)

Reply:
Done [here](https://github.com/antaz/llm/pull/38/commits/8f70d8eed81c108241be21e130f6530d4f6fb107)